packages/sit-onyx/src/components/OnyxDataGrid/features/filtering/filtering.ts

```diff
@@ -0,0 +1,162 @@
+    const filters = ref<Partial<Record<keyof TEntry, string>>>({});
+    const { t } = injectI18n();
+    const config = computed(() => toValue(options?.columns));
+
+    const isFilterEnabled = computed(() => (col: keyof TEntry) => {
+      return !config.value || config.value?.[col]?.enabled === true;
+    });
+
+    const filterData = (data: Readonly<TEntry>[]) => {
+      const filteredData = data.filter((entry) =>
+        Object.entries(filters.value).every(([col, value]) => {
+          const columnOptions = config.value?.[col as keyof TEntry];
+          const filterOpts = {
+            ...options?.filterConfig,
+            ...columnOptions?.filterConfig,
+          };
+
+          if (columnOptions?.filterFunc) {
+            return columnOptions.filterFunc(
+              entry[col as keyof TEntry],
+              value as TEntry[keyof TEntry],
+            );
+          }
+          let filterValue = value as string;
+          let entryValue = entry[col as keyof TEntry]?.toString() || "";
+
+          if (filterOpts?.trimWhitespace) {
+            entryValue = entryValue.replace(/\s+/g, "");
+          }
+
+          if (!filterOpts?.caseSensitive) {
+            entryValue = entryValue.toLowerCase();
+            filterValue = filterValue.toLowerCase();
+          }
+
+          if (filterOpts?.exactMatch) {
+            return entryValue === filterValue;
+          }
+
+          return filterOpts?.searchFromStart
+            ? entryValue.startsWith(filterValue)
+            : entryValue.includes(filterValue);
+        }),
+      );
+      return filteredData;
+    };
+
+    const clearFilter = (column: keyof DataGridEntry) => {
+      filters.value[column] = "";
+    };
+
+    // sync filters with user provided config
+    watchEffect(() => {
+      if (!config.value) return;
+      Object.entries(config.value).forEach(([column, columnOptions]) => {
+        if (columnOptions?.filter) {
+          filters.value[column] = columnOptions.filter;
+        }
+      });
+    });
+
+    const getMenuItem = (column: keyof DataGridEntry) => {
+      let inputValue = filters.value[column] || "";
+
+      const handleKeyDown = (e: KeyboardEvent) => {
+        if (e.key === "Enter") {
+          filters.value[column] = inputValue;
+        }
+      };
+
+      return h(OnyxMiniSearch, {
+        label: column as string,
+        class: "onyx-filter-search",
+        style: {
+          minWidth: "11rem",
+        },
+        hideLabel: true,
+        placeholder: t.value(`dataGrid.head.filtering.menu.placeholder`),
+        modelValue: inputValue,
+        autofocus: true,
+        "onUpdate:modelValue": (value: string) => {
+          inputValue = value;
+        },
+        onKeydown: handleKeyDown,
+        onClick: (e: MouseEvent) => {
+          e.preventDefault();
+          e.stopPropagation();
+        },
+      } as OnyxMiniSearchProps);
+    };
+
+    return {
+      name: FILTERING_FEATURE,
+      watch: [filters],
```

Comment:
```suggestion
      watch: [filters, config],
```

We should also watch the user provided config